packages/next-app/src/components/OwnedPool.tsx

```diff
@@ -2,23 +2,46 @@ import { Badge, Button, HStack, Text, VStack } from "@chakra-ui/react";
 
 interface IOwnedPool {
   onClick: () => void;
-  address: string;
-  type: string;
-  ratio?: string;
-  name: string;
+  address?: string;
+  poolType?: string | null | undefined;
+  name?: string | null | undefined;
+  id?: string | null | undefined;
+  tokens?:
+    | {
+        symbol: string;
+        weight?: unknown;
+      }[]
+    | null
+    | undefined;
   isSelected: boolean;
 }
 
+function truncateAddress(address: string | undefined) {
+  if (!address) return;
+  const match = address.match(
+    /^(0x[a-zA-Z0-9]{10})[a-zA-Z0-9]+([a-zA-Z0-9]{4})$/
+  );
+  if (!match) return address;
+  return `${match[1]}…${match[2]}`;
+}
+
 export function OwnedPool({
   onClick,
   address,
-  type,
-  ratio,
+  poolType,
   name,
   isSelected,
+  tokens,
 }: IOwnedPool) {
   const backgroundColor = isSelected ? "blue.800" : "gray.800";
 
+  const poolName =
+    poolType === "Weighted" ? tokens!.map((obj) => obj.symbol).join("/") : name;
+  const weights =
+    poolType === "Weighted"
+      ? tokens!.map((obj) => (Number(obj.weight) * 100).toFixed()).join("/")
+      : null;
```

Comment:
any reason to mark those as a non-null assertion? If there's a scenario where `tokens` is nullish I'd say we need to make the component handle that case, otherwise this could be creating a bug (although it sounds to me more like an impossible scenario caused by the way we defined the component types here).

Reply:
I forgot to remove it. I'll check if the tokens exist before creating these variables.